docs/Tutorials/OpenAPI/OpenAPI.md

```diff
@@ -0,0 +1,1044 @@
+
+```powershell
+New-PodeOAIntProperty -Name 'userId' -Object
+```
+
+In a response as JSON, this could look as follows:
+
+```json
+{
+    "userId": 0
+}
+```
+
+Furthermore, you can also supply both `-Array` and `-Object` switches:
+
+```powershell
+New-PodeOAIntProperty -Name 'userId' -Object -Array
+```
+
+This wil result in something like the following JSON:
+
+```json
+{
+    "userId": [ 0, 1, 2 ]
+}
+```
+
+#### Complex
+
+Unlike the `-Object` switch that simply converts a single property into an object, the [`New-PodeOAObjectProperty`] function can combine and convert multiple properties.
+
+For example, the following will create an object using an Integer, String, and a Boolean:
+
+Legacy Definition
+
+```powershell
+New-PodeOAObjectProperty -Properties (
+    (New-PodeOAIntProperty -Name 'userId'),
+    (New-PodeOAStringProperty -Name 'name'),
+    (New-PodeOABoolProperty -Name 'enabled')
+)
+```
+
+Using piping (new in Pode 2.10)
+
+```powershell
+New-PodeOAIntProperty -Name 'userId'| New-PodeOAStringProperty -Name 'name'|
+   New-PodeOABoolProperty -Name 'enabled' |New-PodeOAObjectProperty
+```
+
+As JSON, this could look as follows:
+
+```json
+{
+    "userId": 0,
+    "name": "Gary Goodspeed",
+    "enabled": true
+}
+```
+
+You can also supply the `-Array` switch to the [`New-PodeOAObjectProperty`] function. This will result in an array of objects. For example, if we took the above:
+
+```powershell
+New-PodeOAIntProperty -Name 'userId'| New-PodeOAStringProperty -Name 'name'|
+   New-PodeOABoolProperty -Name 'enabled' |New-PodeOAObjectProperty -Array
+```
+
+As JSON, this could look as follows:
+
+```json
+[
+    {
+        "userId": 0,
+        "name": "Gary Goodspeed",
+        "enabled": true
+    },
+    {
+        "userId": 1,
+        "name": "Kevin",
+        "enabled": false
+    }
+]
+```
+
+You can also combine objects into other objects:
+
+```powershell
+$usersArray = New-PodeOAIntProperty -Name 'userId'| New-PodeOAStringProperty -Name 'name'|
+   New-PodeOABoolProperty -Name 'enabled' |New-PodeOAObjectProperty -Array
+
+New-PodeOAObjectProperty -Properties @(
+    (New-PodeOAIntProperty -Name 'found'),
+    $usersArray
+)
+```
+
+As JSON, this could look as follows:
+
+```json
+{
+    "found": 2,
+    "users": [
+        {
+            "userId": 0,
+            "name": "Gary Goodspeed",
+            "enabled": true
+        },
+        {
+            "userId": 1,
+            "name": "Kevin",
+            "enabled": false
+        }
+    ]
+}
+```
+
+### oneOf, anyOf and allOf Keywords
+
+OpenAPI 3.x provides several keywords which you can use to combine schemas. You can use these keywords to create a complex schema or validate a value against multiple criteria.
+
+* oneOf – validates the value against exactly one of the sub-schemas
+* allOf – validates the value against all the sub-schemas
+* anyOf – validates the value against any (one or more) of the sub-schemas
```

Comment:
I think the dashes (-) here (after the oneOf, allOf, anyOf) might be MS Word long dashes, as they render as "?" for me. 

Might be worth deleting and directly replacing with a dash again 😆

Reply:
yes you are right. Probably some copy and paste :)